### PR TITLE
[Playlist][M3U] Fix long line reading

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

Since each platform read method require fixed buffering
i implemented a chunked buffer read method by chucks of 1024 seem reasonable value to me

## Motivation and context

i was testing InputStream Adaptive with STRM files
an noticed that STRM with very long properties value was not read correctly
so the content of the string was being truncated

are corner cases, but anyway the current file line reading method is not correct

## How has this been tested?

this is a example [VeryLongProp.strm.txt](https://github.com/user-attachments/files/17269570/VeryLongProp.strm.txt)
not playable but can be used to debug `ReadString` method
where a property line is higher than 4095 chars

## What is the effect on users?

Can use STRM with very long properties values and so play videos correctly

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/filesystem/File.cpp [hunk 1]
```diff
@@ -765,6 +765,68 @@ int64_t CFile::GetPosition() const
...
+bool XFILE::CFile::ReadString(std::vector<char>& line)
+{
+  if (!m_pFile)
+    return false;
+
+  line.clear();
+
+  if (m_pBuffer)
+  {
+    typedef CFileStreamBuffer::traits_type traits;
```
Comment (enen92) [inline xbmc/filesystem/File.cpp:777]: `using traits = CFileStreamBuffer::traits_type;`

File: xbmc/filesystem/File.cpp [hunk 2]
```diff
@@ -765,6 +765,68 @@ int64_t CFile::GetPosition() const
...
+
+      line.emplace_back(traits::to_char_type(aByte));
+    }
+
+    return true;
+  }
+
+  try
+  {
+    // Read by buffer chucks until to EOL or EOF
```
Comment (enen92) [inline xbmc/filesystem/File.cpp:809]: typo chunks

File: xbmc/filesystem/File.cpp (hunk 1 above)
Comment (CastagnaIT) [reply xbmc/filesystem/File.cpp]: done

**Comment by CastagnaIT** [comment]
jenkins build this please

File: xbmc/filesystem/File.cpp [hunk 3]
```diff
@@ -765,6 +765,68 @@ int64_t CFile::GetPosition() const
...
+    // Read by buffer chuncks until to EOL or EOF
+    do
+    {
+      char bufferLine[1025];
+      if (!m_pFile->ReadString(bufferLine, 1024)) // EOF or error
+        return !line.empty();
+
+      const size_t length = std::strlen(bufferLine);
+      line.reserve(line.size() + length);
+      std::copy(bufferLine, bufferLine + length, std::back_inserter(line));
```
Comment (neo1973) [inline xbmc/filesystem/File.cpp:818]:
```suggestion
      line.insert(line.end(), bufferLine, bufferLine + length);
```

File: xbmc/filesystem/File.cpp [hunk 4]
```diff
@@ -765,6 +765,68 @@ int64_t CFile::GetPosition() const
...
+    return true;
+  }
+
+  try
+  {
+    // Read by buffer chuncks until to EOL or EOF
+    do
+    {
+      char bufferLine[1025];
+      if (!m_pFile->ReadString(bufferLine, 1024)) // EOF or error
```
Comment (neo1973) [inline xbmc/filesystem/File.cpp:813]:
```suggestion
      char bufferLine[1025];
      if (!m_pFile->ReadString(bufferLine, sizeof(bufferLine))) // EOF or error
```
If you only want to read 1023 bytes from the file make `bufferLine` 1024. Either way the extra unused byte is confusing.

File: xbmc/filesystem/File.cpp (hunk 3 above)
Comment (CastagnaIT) [reply xbmc/filesystem/File.cpp]: applied

File: xbmc/filesystem/File.cpp (hunk 4 above)
Comment (CastagnaIT) [reply xbmc/filesystem/File.cpp]: applied

**Review by neo1973 — APPROVED** [review]
Not runtime tested but looks good to me :slightly_smiling_face:

**Comment by CastagnaIT** [comment]
i tested again after the changes, works as expected

**Pull request merged**